### PR TITLE
Update dependencies to resolve peer conflicts

### PR DESCRIPTION
**Pull request by thisisdano**
Update: I've also added a couple USWDS 3.2.0–related changes to this PR:

- Updated README references to 3.2.0
- Updated notifications to include markup changes in 3.2.0

- - - 

This PR updates a few of our dev dependencies to prevent installation warnings and errors related to peer dependency conflicts.

Most of these changes are small version bumps, but this does add a dependency to `react-dom@17.0.2`. This helps npm resolve peer dependencies in subdependencies, and allows us to install the package without conflict warnings or errors. 

The update to `handlebars-helper` resolves 3 `moderate` vulnerabilities. 

**node:** 16.17.1
**npm:** 8.15.0

## Before

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-inspector@5.1.1
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peerOptional react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @storybook/addon-a11y@6.5.5
npm WARN   node_modules/@storybook/addon-a11y
npm WARN     dev @storybook/addon-a11y@"6.5.5" from the root project
npm WARN   38 more (@storybook/addon-actions, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN node_modules/@storybook/addon-actions/node_modules/react-inspector
npm WARN   react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.5
npm WARN   node_modules/@storybook/addon-actions
...
```

`29 vulnerabilities (3 moderate, 26 high)`

## After

No errors or warnings

`26 high severity vulnerabilities`

### CODE REVIEW

**Comment by thisisdano** [comment]
Hm. Circle seems to be a little confused here. I see this as completed in Circle (https://app.circleci.com/pipelines/github/uswds/uswds?branch=dw-dependencies-3.2.0), but the failure here links to another branch build (`
dw-update-node-lts-16.18.0`) (https://app.circleci.com/pipelines/github/uswds/uswds/4284/workflows/fe36b6ae-684d-4558-a985-fc4cb0ed228e/jobs/8704)

File: package.json [hunk 1]
```diff
@@ -101,6 +101,7 @@
   "homepage": "https://github.com/uswds/uswds#readme",
   "dependencies": {
     "classlist-polyfill": "1.0.3",
+    "install": "0.13.0",
```
Comment (mejiaj) [inline package.json:104]: What does this new dependency do? Everything else looks good.

**Review by mejiaj — COMMENTED** [review]
Just a question on the `install` dependency.

File: package.json (hunk 1 above)
Comment (thisisdano) [reply package.json:104]: huh. that looks like an error

**Review by mejiaj — APPROVED** [review]

**Pull request merged**